packages/backend/src/api/1/handlers/auth.js

```diff
@@ -1,3 +1,33 @@
+import crypto from 'crypto';
 import express from 'express';
+import passport from 'passport';
 
-module.exports = express.Router();
+const router = express.Router();
+
+router.get('/reddit/callback', (req, res, next) => {
+  // Check for origin via state token
+  if (req.query.state === req.session.state)
```

Comment:
Actually from what I see, you can extract it into a middleware chain for that request and the code will get a little more readable imo